### PR TITLE
TELCODOCS-843/980: Ecoprojects documentation changes - New concept module / Content restructure

### PR DESCRIPTION
**Pull request by ogradyp**
https://issues.redhat.com/browse/TELCODOCS-843 D/S Docs: [TELCODOCS-819](https://issues.redhat.com//browse/TELCODOCS-819) Ecoprojects documentation changes - New concept module

https://issues.redhat.com/browse/TELCODOCS-980 D/S Docs: [TELCODOCS-819](https://issues.redhat.com//browse/TELCODOCS-819) Ecoprojects documentation changes - Content restructure

Applies to OpenShift version : 4.12 +

New concept module
Preview: [About Remediation, Fencing and Maintencance](https://53866--docspreview.netlify.app/openshift-enterprise/latest/nodes/nodes/ecosystems/eco-about-remediation-fencing-maintenance.html)

Content restructure
Preview:  [Remediation, Fencing and Maintencance](https://53866--docspreview.netlify.app/openshift-enterprise/latest/nodes/nodes/ecosystems/eco-about-remediation-fencing-maintenance.html)

Dev review complete/approved by @slintes
Dev review complete/approved by @mshitrit 
QE review complete/approved by  @prabinovRedhat
Peer review complete/approved by  @mjpytlak

Thank you.

### CODE REVIEW

**Comment by slintes** [comment]
nice! lgtm

**Comment by mshitrit** [comment]
lgtm :+1:

**Comment by prabinovRedhat** [comment]
lgtm

File: nodes/nodes/ecosystems/eco-node-health-check-operator.adoc [hunk 1]
```diff
@@ -25,9 +25,9 @@ include::modules/eco-node-health-check-operator-creating-node-health-check.adoc[
...
 
 [id="additional-resources-nhc-operator-installation"]
 == Additional resources
-* xref:../../operators/admin/olm-upgrading-operators.adoc#olm-changing-update-channel_olm-upgrading-operators[Changing the update channel for an Operator]
-* The Node Health Check Operator is supported in a restricted network environment. For more information, see xref:../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
+* xref:../../../operators/admin/olm-upgrading-operators.adoc#olm-changing-update-channel_olm-upgrading-operators[Changing the update channel for an Operator]
+* The Node Health Check Operator is supported in a restricted network environment. For more information, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
```
Comment (mjpytlak) [inline nodes/nodes/ecosystems/eco-node-health-check-operator.adoc:33]: I struggle getting around this myself, but Additional resources can only contain a list of bulleted links, no text. IMO, I think the link text you use stands on its own just fine.

[1] https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=JUP&title=Jupiter+Source+files+readiness+Checklist#JupiterSourcefilesreadinessChecklist-Additionalresourcesspecificrequirements

File: nodes/nodes/ecosystems/eco-self-node-remediation-operator.adoc [hunk 2]
```diff
@@ -13,25 +13,25 @@ include::modules/eco-self-node-remediation-operator-about.adoc[leveloffset=+1]
 include::modules/eco-self-node-remediation-about-watchdog.adoc[leveloffset=+2]
 
 .Additional resources
```
Comment (mjpytlak) [inline nodes/nodes/ecosystems/eco-self-node-remediation-operator.adoc:15]:
```suggestion
[role="_additional-resources"]
.Additional resources
```

File: nodes/nodes/ecosystems/eco-self-node-remediation-operator.adoc [hunk 3]
```diff
@@ -13,25 +13,25 @@ include::modules/eco-self-node-remediation-operator-about.adoc[leveloffset=+1]
...
+To collect debugging information about the Self Node Remediation Operator, use the `must-gather` tool. For information about the `must-gather` image for the Self Node Remediation Operator, see xref:../../../support/gathering-cluster-data.adoc#gathering-data-specific-features_gathering-cluster-data[Gathering data about specific features].
 
 [role="_additional-resources"]
 [id="additional-resources-self-node-remediation-operator-installation"]
 == Additional resources
 
-* The Self Node Remediation Operator is supported in a restricted network environment. For more information, see xref:../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
-* xref:../../operators/admin/olm-deleting-operators-from-cluster.adoc#olm-deleting-operators-from-a-cluster[Deleting Operators from a cluster]
+* The Self Node Remediation Operator is supported in a restricted network environment. For more information, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
```
Comment (mjpytlak) [inline nodes/nodes/ecosystems/eco-self-node-remediation-operator.adoc:36]: Same comment as previous. No text in these sections.

File: nodes/nodes/ecosystems/eco-machine-health-checks.adoc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+:_content-type: ASSEMBLY
+[id="machine-health-checks"]
+= Remediating nodes with Machine Health Checks
+include::_attributes/common-attributes.adoc[]
+:context: machine-health-checks
+
+toc::[]
+
```
Comment (mjpytlak) [inline nodes/nodes/ecosystems/eco-machine-health-checks.adoc:8]: Please add a sentence or a short introductory paragraph to serve as the abstract.

**Review by mjpytlak — COMMENTED** [review]
Nice work here @ogradyp. Nothing major, but there are a handful of items that require an update.

File: modules/eco-self-node-remediation-operator-configuring.adoc [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+<4> Specify the timeout duration to check connectivity with each API server. When this duration elapses, the Operator starts remediation. The timeout duration must be more than or equal to 10 milliseconds.
+<5> Specify the frequency to check connectivity with each API server. The timeout duration must be more than or equal to 1 second.
+<6> Specify a threshold value. After reaching this threshold, the node starts contacting its peers. The threshold value must be more than or equal to 1 second.
+<7> Specify the duration of the timeout for the peer to connect the API server. The timeout duration must be more than or equal to 10 milliseconds.
+<8> Specify the duration of the timeout for establishing connection with the peer. The timeout duration must be more than or equal to 10 milliseconds.
+<9> Specify the duration of the timeout to get a response from the peer. The timeout duration must be more than or equal to 10 milliseconds.
+<10> Specify the frequency to update peer information, such as IP address. The timeout duration must be more than or equal to 10 seconds.
```
[the remark this answers is not in the document]
Comment (ogradyp) [reply modules/eco-self-node-remediation-operator-configuring.adoc]: Thanks @mjpytlak I've applied your feedback.

File: modules/eco-self-node-remediation-operator-configuring.adoc [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+[id="understanding-self-node-remediation-remediation-template-config_{context}"]
+== Understanding the Self Node Remediation Template configuration
+
+The Self Node Remediation Operator also creates the `SelfNodeRemediationTemplate` Custom Resource Definition (CRD). This CRD defines the remediation strategy for the nodes. The following remediation strategies are available:
+
+`ResourceDeletion`:: This remediation strategy removes the pods and associated volume attachments on the node rather than the node object. This strategy helps to recover workloads faster. `ResourceDeletion` is the default remediation strategy.
+
+`NodeDeletion`:: This remediation strategy is deprecated and will be removed in a future release. In the current release, the `ResourceDeletion` strategy is used even if the `NodeDeletion` strategy is selected.
+
+
```
[the remark this answers is not in the document]
Comment (ogradyp) [reply modules/eco-self-node-remediation-operator-configuring.adoc]: Thanks @mjpytlak I've applied your feedback.

File: modules/eco-self-node-remediation-operator-configuring.adoc [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+The Self Node Remediation Operator creates the following `SelfNodeRemediationTemplate` CR for the strategy:
+
+* `self-node-remediation-resource-deletion-template`, which the `ResourceDeletion` remediation strategy uses
+//* `self-node-remediation-node-deletion-template`, which the `NodeDeletion` remediation strategy uses
+
```
[the remark this answers is not in the document]
Comment (ogradyp) [reply modules/eco-self-node-remediation-operator-configuring.adoc:77]: Thanks @mjpytlak I've applied your feedback.

File: modules/eco-self-node-remediation-operator-installation-web-console.adoc [hunk 8]
```diff
@@ -8,6 +8,12 @@
 
 You can use the {product-title} web console to install the Self Node Remediation Operator.
 
+[NOTE]
+====
+The Node Health Check Operator also installs the Self Node Remediation Operator as a default remediation provider.
+====
+
+
```
[the remark this answers is not in the document]
Comment (ogradyp) [reply modules/eco-self-node-remediation-operator-installation-web-console.adoc]: Thanks @mjpytlak I've applied your feedback.

File: nodes/nodes/ecosystems/eco-about-remediation-fencing-maintenance.adoc [hunk 9]
```diff
@@ -0,0 +1,39 @@
+:_content-type: ASSEMBLY
+[id="about-remediation-fencing-maintenance"]
+= About node remediation, fencing, and maintenance
+include::_attributes/common-attributes.adoc[]
+:context: about-node-remediation-fencing-maintenance
+
+toc::[]
+
+Hardware is imperfect and software contains bugs. When node-level failures, such as, the kernel hangs or network interface controllers (NICs) fail, the work required from the cluster does not decrease, and workloads from affected nodes need to be restarted somewhere. However, some workloads, such as, ReadWriteOnce (RWO) volumes and StatefulSets, may require, at-most-one semantics.
```
[the remark this answers is not in the document]
Comment (ogradyp) [reply nodes/nodes/ecosystems/eco-about-remediation-fencing-maintenance.adoc]: Thanks @mjpytlak I've applied your feedback.

File: nodes/nodes/ecosystems/eco-about-remediation-fencing-maintenance.adoc [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+Failures affecting these workloads risk data loss and/or corruption. It is important to ensure that the node reaches a safe state, known as `fencing` before initiating recovery of the workload, known as `remediation` and ideally, recovery of the node also.
```
[the remark this answers is not in the document]
Comment (ogradyp) [reply nodes/nodes/ecosystems/eco-about-remediation-fencing-maintenance.adoc]: Thanks @mjpytlak I've applied your feedback.

File: nodes/nodes/ecosystems/eco-about-remediation-fencing-maintenance.adoc [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+It is not always practical to depend on administrator intervention to confirm the true status of the nodes and workloads. To facilitate such intervention, OCP provides multiple components for the automation of failure detection, fencing and remediation.
```
[the remark this answers is not in the document]
Comment (ogradyp) [reply nodes/nodes/ecosystems/eco-about-remediation-fencing-maintenance.adoc]: Thanks @mjpytlak I've applied your feedback.

File: nodes/nodes/ecosystems/eco-node-health-check-operator.adoc (hunk 1 above)
Comment (ogradyp) [reply nodes/nodes/ecosystems/eco-node-health-check-operator.adoc]: Thanks @mjpytlak I've applied your feedback.

File: nodes/nodes/ecosystems/eco-self-node-remediation-operator.adoc (hunk 2 above)
Comment (ogradyp) [reply nodes/nodes/ecosystems/eco-self-node-remediation-operator.adoc:16]: Thanks @mjpytlak I've applied your feedback.

File: nodes/nodes/ecosystems/eco-self-node-remediation-operator.adoc (hunk 3 above)
Comment (ogradyp) [reply nodes/nodes/ecosystems/eco-self-node-remediation-operator.adoc]: Thanks @mjpytlak I've applied your feedback.

File: nodes/nodes/ecosystems/eco-machine-health-checks.adoc (hunk 4 above)
Comment (ogradyp) [reply nodes/nodes/ecosystems/eco-machine-health-checks.adoc:8]: Thanks @mjpytlak I've applied your feedback.

File: _topic_maps/_topic_map.yml [hunk 12]
```diff
@@ -2098,12 +2098,19 @@ Topics:
     File: nodes-nodes-managing-max-pods
   - Name: Using the Node Tuning Operator
     File: nodes-node-tuning-operator
-  - Name: Remediating nodes with the Self Node Remediation Operator
-    File: eco-self-node-remediation-operator
-  - Name: Deploying node health checks by using the Node Health Check Operator
-    File: eco-node-health-check-operator
-  - Name: Using the Node Maintenance Operator to place nodes in maintenance mode
-    File: eco-node-maintenance-operator
+  - Name: Remediation, Fencing, and Maintentance
```
Comment (jeana-redhat) [inline _topic_maps/_topic_map.yml:2101]: Checking the capitalization here, should it be as follows? Also small typo in "maintenance" :)
```suggestion
  - Name: Remediation, fencing, and maintenance
```

File: _topic_maps/_topic_map.yml [hunk 13]
```diff
@@ -2098,12 +2098,19 @@ Topics:
...
+    Dir: ecosystems
+    Topics:
+    - Name: About node remediation, fencing, and maintentance
```
Comment (jeana-redhat) [inline _topic_maps/_topic_map.yml:2104]:
```suggestion
    - Name: About node remediation, fencing, and maintenance
```

File: _topic_maps/_topic_map.yml [hunk 14]
```diff
@@ -2098,12 +2098,19 @@ Topics:
...
+      File: eco-about-remediation-fencing-maintenance
+    - Name: Using Self Node Remediation
+      File: eco-self-node-remediation-operator
+    - Name: Remediating nodes with Machine Health Checks
+      File: eco-machine-health-checks
+    - Name: Remediating nodes with Node Health Checks
+      File: eco-node-health-check-operator
+    - Name: Placing nodes in maintenance mode with Node Maintenance Operator
```
Comment (jeana-redhat) [inline _topic_maps/_topic_map.yml:2112]:
```suggestion
    - Name: Placing nodes in maintenance mode with the Node Maintenance Operator
```

**Review by jeana-redhat — COMMENTED** [review]
Generally LGTM, but did see a couple small typos in the nav titles while scanning the structure in my pre-merge review.

File: _topic_maps/_topic_map.yml (hunk 13 above)
Comment (ogradyp) [reply _topic_maps/_topic_map.yml:2104]: Thanks @jeana-redhat I've committed your suggestion.

**Comment by mburke5678** [comment]
Closed in favor of: https://github.com/openshift/openshift-docs/pull/54678

**Comment by ogradyp** [comment]
Thank you @mburke5678. Much appreciated taking this original, and merging it's content via PR #54678.

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.13` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.12. And any PR going into `main` must also target the latest version branch (enterprise-4.13).

If the update in your PR does NOT apply to version 4.13 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.14` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.13. And any PR going into `main` must also target the latest version branch (enterprise-4.14).

If the update in your PR does NOT apply to version 4.14 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Pull request closed without merging**